source/developer/slash-commands.rst

```diff
@@ -128,54 +89,184 @@ If you'd like to build your own integration that uses slash commands, you can fo
+ .. code-block::
 
-2. Including ``@username`` will trigger a mention notification for the person with the specified username, and channels can be mentioned by including *@channel* or *<!channel>*. For example:  ``{"text": "<!channel> this is a notification"}`` would create a message that mentions *@channel*
+    payload={"text": "<!channel> this is a notification."}
 
-3. If the text in a response is longer than 4000 characters, the message is split into multiple consecutive posts, each within the 4000 character limit.
+Override the username
+~~~~~~~~~~~~~~~~~~~~~
+
+In addition to specifying the response username when setting up your slash command, you can also override it by specifying a ``username`` parameter in your JSON payload.
+
+For example, to send the message as a ``webhook-bot``, use the following payload.
+
+  .. code-block::
+
+    payload={"username": "webhook-bot", "text": "Hello, this is some text\nThis is more text. :tada:"}
+  
+.. image:: ../images/incoming_webhooks_override_username.png
+  :width: 50 px
+
+To prevent malicious users from trying to perform `phishing attacks <https://en.wikipedia.org/wiki/Phishing>`_ a *BOT* indicator appears next to posts coming from webhooks regardless of what username is specified.
+
+  .. note::
+    `Enable integrations to override usernames <https://docs.mattermost.com/administration/config-settings.html#enable-integrations-to-override-usernames>`_ must be set to `true` in `config.json` to override usernames. Enable them from **System Console > Integrations > Custom Integrations** or ask your System Administrator. If not enabled, the username is set to `webhook`.
+
+Override the profile picture
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Similary to the username, you can also override the profile picture by specifying an ``icon_url`` parameter in your JSON payload.
+
+For example, you can use the following payload to override the profile picture to use the image located at http://example.com/somecoolimage.jpg.
+
+  .. code-block::
+
+    payload={"icon_url": "http://example.com/somecoolimage.jpg", "text": "Hello, this is some text\nThis is more text. :tada:"}
+
+  .. note::
+    `Enable integrations to override profile picture icons <https://docs.mattermost.com/administration/config-settings.html#enable-integrations-to-override-profile-picture-icons>`_ must be set to `true` in `config.json` to override usernames. Enable them from **System Console > Integrations > Custom Integrations** or ask your System Administrator. If not enabled, the icon of the creator of the webhook URL is used to post messages.
+
+Tips and Best Practices
+------------------------
+
+1. Slash commands are designed to post messages. For other actions such as channel creation, use the `Mattermost APIs <../developer/api.html>`_.
```

Comment:
Maybe change the first sentence to "Slash commands are designed to easily allow you to post messages. For other actions such as channel creation, you must also use the Mattermost APIs"? Right now, it sounds a bit like you can't have them do other things